lib/perfdata/influxdbchecktask.cpp

```diff
@@ -0,0 +1,149 @@
+
+	double queriesCritical = MacroProcessor::ResolveMacros("$influxdbwriter_queries_critical$", resolvers, checkable->GetLastCheckResult(),
+		&missingQueriesCritical, MacroProcessor::EscapeCallback(), resolvedMacros, useResolvedMacros);
+
+	double pendingDataBuffersWarn = MacroProcessor::ResolveMacros("$influxdbwriter_pending_queries_warning$", resolvers, checkable->GetLastCheckResult(),
+		&missingPendingDataBuffersWarn, MacroProcessor::EscapeCallback(), resolvedMacros, useResolvedMacros);
+
+	double pendingDataBuffersCrit = MacroProcessor::ResolveMacros("$influxdbwriter_pending_queries_critical$", resolvers, checkable->GetLastCheckResult(),
+		&missingPendingDataBuffersCrit, MacroProcessor::EscapeCallback(), resolvedMacros, useResolvedMacros);
+
+	if (resolvedMacros && !useResolvedMacros)
+		return;
+
+	if (influxdbWriterName.IsEmpty()) {
+		ReportInfluxdbCheck(checkable, commandObj, cr, "Attribute 'influxdbwriter_name' must be set.");
+		return;
+	}
+
+	InfluxdbWriter::Ptr influxdbWriter = InfluxdbWriter::GetByName(influxdbWriterName);
+
+	if (!influxdbWriter) {
+		ReportInfluxdbCheck(checkable, commandObj, cr, "InfluxdbWriter '" + influxdbWriterName + "' object does not exist.");
+		return;
+	}
+
+	if (influxdbWriter->IsPaused()) {
+		ReportInfluxdbCheck(checkable, commandObj, cr, "InfluxdbWriter is temporarily disabled on this cluster instance.", state);
+		return;
+	}
+
+	auto errorMessage = influxdbWriter->GetLastErrorMessage();
+	if (!errorMessage.IsEmpty()) {
+		ReportInfluxdbCheck(checkable, commandObj, cr, errorMessage, ServiceCritical);
+		return;
+	}
+
+	double wqItemPerSec = influxdbWriter->GetQueryCount(60) / 60.0;
+	double pendingDataBuffers = influxdbWriter->GetPendingDataBuffers();
+	std::ostringstream msgBuffer;
+
+	msgBuffer << "Queries per second: " << std::fixed << std::setprecision(3) << wqItemPerSec << ".";
+
+	/* Check whether the thresholds have been defined and match. */
+	if (missingQueriesCritical.IsEmpty() && wqItemPerSec < queriesCritical) {
+		msgBuffer << " " << wqItemPerSec << " queries/s lower than critical threshold (" << queriesCritical << " queries/s).";
+
+		state = ServiceCritical;
+	} else if (missingQueriesWarning.IsEmpty() && wqItemPerSec < queriesWarning) {
+		msgBuffer << " " << wqItemPerSec << " queries/s lower than warning threshold (" << queriesWarning << " queries/s).";
+
+		state = ServiceWarning;
+	}
+
+	if (missingPendingDataBuffersCrit.IsEmpty() && pendingDataBuffers > pendingDataBuffersCrit) {
+		msgBuffer << " " << pendingDataBuffers << " pending data buffers greater than critical threshold ("
+			<< pendingDataBuffersCrit << " queries).";
+
+		state = ServiceCritical;
+	} else if (missingPendingDataBuffersWarn.IsEmpty() && pendingDataBuffers > pendingDataBuffersWarn) {
+		msgBuffer << " " << pendingDataBuffers << " pending data buffers greater than warning threshold ("
+			<< pendingDataBuffersWarn << " queries).";
+
+		state = ServiceWarning;
```

Comment:
This might overwrite a ServiceCritical set in line 121. (This bug also exists in the IDO check.)

Reply:
@Yonas-net Open a fix-PR for the IDO check. Once it's been merged, we'll come back to this construction area here.